MerlinAU.sh

```diff
@@ -1214,6 +1216,36 @@ _GetLatestFWUpdateVersionFromRouter_()
    echo "$newVersionStr" ; return "$retCode"
 }
 
+##---------------------------------------##
+## Added by ExtremeFiretop [2024-Apr-18] ##
+##---------------------------------------##
+_GetFirmwareVariantFromRouter_()
+{
+   local retCode=0  newVersionStr
+
+   newVersionStr="$(nvram get webs_state_info | sed 's/_/./g')"
+   if [ $# -eq 0 ] || [ -z "$1" ]
+   then
+       newVersionStr="$(echo "$newVersionStr" | awk -F '-' '{print $1}')"
+   fi
+
+   ##FOR TESTING/DEBUG ONLY##
+   if false # Change to true for forcing GNUton flag
+   then 
+      isGNUtonFW=true
+   else
+      # Check if the version string contains "gnuton"
+      if echo "$newVersionStr" | grep -iq "gnuton"; then
+          isGNUtonFW=true
+      else
+          isGNUtonFW=false
+      fi
+   fi
+
+   [ -z "$isGNUtonFW" ] && retCode=1
```

Comment:
Delete line#**1245**. It's not needed because the variable "`isGNUtonFW`" would never be empty.